py3-graph-tool.yaml

```diff
@@ -0,0 +1,124 @@
+package:
+  name: py3-graph-tool
+  version: 2.97
+  epoch: 0
+  description: graph-tool is an efficient Python module for manipulation and statistical analysis of graphs (a.k.a. networks).
+  resources:
+    cpu: 65
+    memory: 128Gi
+  copyright:
+    - license: GPL-3.0-only
+  target-architecture:
+    - x86_64
+
+vars:
+  pypi-package: graph-tool
+  import: graph_tool
+
+data:
+  # TODO: add other supported versions
+  - name: py-versions
+    items:
+      3.13:
+
+environment:
+  contents:
+    packages:
+      - autoconf
+      - automake
+      - bash
+      - boost-dev
+      - build-base
+      - busybox
+      - ca-certificates-bundle
+      - cairo-dev
+      - cairomm-dev
+      - cgal-dev
+      - eigen-dev
+      - expat-dev
+      - freetype-dev
+      - gmp-dev
+      - libsigcplusplus-dev
+      - libtool
+      - mpfr-dev
+      - pkgconf-dev
+      - py3.13-cairo
+      - py3.13-numpy
+      - py3.13-scipy
+      - python3-dev
+      - sparsehash-dev
+      - zlib-dev
+  environment:
+    CAIROMM_LIBS: "/usr/lib/cairomm-1.16"
+    CAIROMM_CFLAGS: "-Wno-error"
+    PYTHON_CPPFLAGS: "-I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/python3.13"
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://git.skewed.de/count0/graph-tool
+      tag: release-${{package.version}}
+      expected-commit: cd15650ec9fb6807b4536b523a059354f231fcb1
+
+subpackages:
+  - range: py-versions
+    name: py${{range.key}}-${{vars.pypi-package}}
+    description: python${{range.key}} version of ${{vars.pypi-package}}
+    dependencies:
+      provides:
+        - graph-tool=${{package.full-version}}
+      runtime:
+        - py${{range.key}}-numpy
+        - py${{range.key}}-scipy
+    pipeline:
+      - runs: |
+          ./autogen.sh
+          CXXFLAGS=${CXXFLAGS/-Wp,-D_GLIBCXX_ASSERTIONS/}
+          export CXXFLAGS="$CXXFLAGS -O3"
+      - uses: autoconf/configure
+        with:
+          opts: |
+            --enable-openmp \
+            --includedir=/usr/lib/cairomm-1.16 \
+            PYTHON=python${{range.key}}
+      - runs: |
+          ln -s /usr/lib/cairomm-1.16/include/cairommconfig.h /usr/include/cairommconfig.h
+          ln -s /usr/include/cairomm-1.16/cairomm/ /usr/include/cairomm
+          ln -s /usr/lib/sigc++-3.0/include/sigc++config.h /usr/include/sigc++config.h
+          ln -s /usr/include/sigc++-3.0/sigc++/ /usr/include/sigc++
+          export CPPFLAGS="$CPPFLAGS -I/usr/lib/cairomm-1.16/include -I/usr/include/cairomm-1.16/cairomm/ -I/usr/include/sigc++-3.0/sigc++/"
```

Comment:
not liking this very much but something to do with lots of configure flags and environment variables. I tried using `-I` and `--includedir` and exporting environment variables again and again but all of them were not able to find headers so ended up symlinking them to `/usr/include`

Reply:
this indicates broken packaging; of those includes; and theri pkgconfig file. Instead of doing this every time, it is best to fix those pc files to do the right thing out of the box.